.github/workflows/build.yml

```diff
@@ -122,6 +122,10 @@ jobs:
           cargo test --verbose --color always --no-default-features --features no-std
           # check if there is a conflict between no-std and the default std feature
           cargo test --verbose --color always --features no-std
+          # check no-std compatibility across dependencies
```

Comment:
I wasn't unfortunately. In-crate compiling does catch anything using `std` but apparently not methods unavailable on primitive types. Maybe there is some configuration that would trigger it? I was using `cargo check -p lightning --no-default-features --features=no-std`.

Reply:
This feels like something we should maybe report as a rustc bug?

Reply:
Interestingly, I see the compilation fail with a minimal example in Rust playground:

```rust
#![no_std]

fn main() {
    1.0_f64.log10();
}
```

```
   Compiling playground v0.0.1 (/playground)
error: language item required, but not found: `eh_personality`
  |
  = note: this can occur when a binary crate with `#![no_std]` is compiled for a target where `eh_personality` is defined in the standard library
  = help: you may be able to compile for a target that doesn't need `eh_personality`, specify a target with `--target` or in `.cargo/config`

error: `#[panic_handler]` function required, but not found

error[[E0599]](https://doc.rust-lang.org/stable/error-index.html#E0599): no method named `log10` found for type `f64` in the current scope
 [--> src/main.rs:4:13
](https://play.rust-lang.org/#)  |
4 |     1.0_f64.log10();
  |             ^^^^^ method not found in `f64`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `playground` due to 3 previous errors
```

If I change the empty `lib.rs` file in `no-std-check` to use a minimal example, I only see the last error. 

```rust
#![no_std]

pub fn log(x: f64) -> f64 {
    x.log10()
}      
```

So I wonder if there is something about our configuration that is causing this to not be caught.

Reply:
perhaps the issue is with code that is not reachable from `main()`, but only from tests?

Reply:
ignore my last comment, sounds like you did try it without a call from `main`

Reply:
Right, so it sounds like the bug is that rustc doesn't detect this as an issue if we're building for test/check instead of building for run?

Reply:
To clarify, `rustc` does catch it when using `cd no-std-check && cargo check` with the modified `lib.rs`:

```
    Checking no-std-check v0.1.0 (/Users/jkczyz/src/rust-lightning/no-std-check)
error[E0599]: no method named `log10` found for type `f64` in the current scope
 --> src/lib.rs:4:7
  |
4 |     x.log10()
  |       ^^^^^ method not found in `f64`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: could not compile `no-std-check`

To learn more, run the command again with --verbose.
```

Reply:
Right, so the issue is that some crates in the workspace are not no-std so everything gets built with std?